pulsar/consumer.go

```diff
@@ -140,6 +140,9 @@ type ConsumerOptions struct {
 
 	// A chain of interceptors, These interceptors will be called at some points defined in ConsumerInterceptor interface.
 	Interceptors ConsumerInterceptors
+
+	// MaxReconnectToBroker set the maximum retry number of reconnectToBroker. (default: ultimate)
+	MaxReconnectToBroker *uint
```

Comment:
IMO using `int` type is more friendly,  the default value `0` represents unlimited.

Reply:
> IMO using `int` type is more friendly, the default value `0` represents unlimited.

How aboud `0` represents no retry but nil means retry ultimate times? Then this changes won't break current behavior of reconnectToBroker.

Reply:
I see your point, but it's common to use int for this kind of configuration, such as [go-redis](https://github.com/go-redis/redis/blob/master/options.go#L59) / [nats.go](https://github.com/nats-io/nats.go/blob/master/nats.go#L266). 

* 0 for no-try
* negative number for retry forever
* positive number for retry attempts

As current library is still in pre-1.0, i think it's ok to make a break change.

Reply:
> I see your point, but it's common to use int for this kind of configuration, such as [go-redis](https://github.com/go-redis/redis/blob/master/options.go#L59) / [nats.go](https://github.com/nats-io/nats.go/blob/master/nats.go#L266), we can make the following convention for the config to cope with different scenarios,
> 
> * 0 for no retry
> * negative number for retry forever
> * positive number to set retry attempts
> 
> As the current library is still in pre-1.0, i think it's ok to make a break change.

Totally agree, Do we need to ask for other maintainer's opinions?

Reply:
invite @wolfstudy to review?

Reply:
> invite @wolfstudy to review?

Sounds great!

Reply:
Thanks @jonyhy96,  @shohi proposal LGTM +, can you help fix this?